src/NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1446,12 +1446,21 @@ private bool EvalScript(Script s, TransactionChecker checker, int hashversion)
                                             return SetError(ScriptError.InvalidStackOperation);
 
                                         int nKeysCount = new CScriptNum(this._stack.Top(-i), fRequireMinimal).getint();
-                                        if(nKeysCount < 0 || nKeysCount > 20)
-                                            return SetError(ScriptError.PubkeyCount);
 
-                                        nOpCount += nKeysCount;
-                                        if(nOpCount > 201)
-                                            return SetError(ScriptError.OpCount);
+                                        if (this.Network.Federations != null && lastOpcode?.Code == OpcodeType.OP_FEDERATION)
+                                        {
+                                            if (nKeysCount < 0 || nKeysCount > 150)
+                                                return SetError(ScriptError.PubkeyCount);
+                                        }
+                                        else
+                                        {
+                                            if (nKeysCount < 0 || nKeysCount > 20)
+                                                return SetError(ScriptError.PubkeyCount);
+
+                                            nOpCount += nKeysCount;
```

Comment:
Why? Do we need this?

Reply:
Basically this is a consensus rule; there can only be a maximum of 201 non-push opcodes in any given script. If this was to be changed then I think we need to look at subclassing `ScriptEvaluationContext` so that per-network versions can be injected.

https://bitcoin.stackexchange.com/questions/38230/maximum-number-of-op-codes-in-script